namerd/iface/control-http/src/main/scala/io/buoyant/namerd/iface/HttpControlService.scala

```diff
@@ -341,15 +341,28 @@ class HttpControlService(storage: DtabStore, delegate: Ns => NameInterpreter, na
   private[this] def getBind(ns: String, path: Path, extraDtab: Option[String]): Activity[NameTree[Name.Bound]] =
     extraDtab match {
       case Some(dtab) => delegate(ns).bind(Dtab.read(dtab), path)
-      case _ => bindingCache.get((ns, path))
+      case _ => bindingCache.get((ns, path))._1
     }
 
+  type NsPath = (String, Path)
+  type ActClose = (Activity[NameTree[Name.Bound]], Closable)
+
   private[this] val bindingCache = CacheBuilder.newBuilder()
     .maximumSize(bindingCacheSize)
-    .build[(String, Path), Activity[NameTree[Name.Bound]]](
-      new CacheLoader[(String, Path), Activity[NameTree[Name.Bound]]] {
-        override def load(key: (String, Path)): Activity[NameTree[Bound]] =
-          delegate(key._1).bind(Dtab.empty, key._2)
+    .removalListener(new RemovalListener[NsPath, ActClose] {
+      override def onRemoval(notification: RemovalNotification[(String, Path), (Activity[NameTree[Bound]], Closable)]): Unit = {
+        if (notification.wasEvicted()) {
```

Comment:
I think we want to close the observation for any kind of removal, not just evictions